File: src/main/java/com/squarepolka/digiscope/plot/plotpoint/PlotPoint.java
package com.squarepolka.digiscope.plot.plotpoint;

import java.math.BigDecimal;

public abstract class PlotPoint {

    private PlotPoint previousPoint;
    private PlotPoint nextPlotPoint;
    private BigDecimal timestampSeconds;

    public PlotPoint(BigDecimal timestampSeconds) {
        this.previousPoint = null;
        this.nextPlotPoint = null;
        this.timestampSeconds = timestampSeconds;
    }

    public BigDecimal getTimestampSeconds() {
        return timestampSeconds;
    }

    public BigDecimal getTimestampMilliseconds() {
        return timestampSeconds.multiply(new BigDecimal(1000));
    }

    public String toString() {
        return getTimestampMilliseconds() + "ms";
    }


    // Generated getters and setters
    public PlotPoint getPreviousPoint() {
        return previousPoint;
    }

    public void setPreviousPoint(PlotPoint previousPoint) {
        this.previousPoint = previousPoint;
    }

    public PlotPoint getNextPlotPoint() {
        return nextPlotPoint;
    }

    public void setNextPlotPoint(PlotPoint nextPlotPoint) {
        this.nextPlotPoint = nextPlotPoint;
    }
}

File: src/main/kotlin/plotpoint/PlotPoint.kt
package plotpoint;

import java.math.BigDecimal

open class PlotPoint(val timestampSeconds: BigDecimal = BigDecimal(0), val voltValue: BigDecimal = BigDecimal(0)) {

    var previousPoint: PlotPoint? = null
    var nextPlotPoint: PlotPoint? = null
    var previousLargerPoint: PlotPoint? = null
    var nextSmallerPoint: PlotPoint? = null
    var index: Int = 0
    
    fun getTimestampMilliseconds(): BigDecimal {
        return timestampSeconds.multiply(BigDecimal(1000))
    }

    override
    fun toString(): String {
        val timestampMilliseconds = getTimestampMilliseconds()
        return "${index} : ${timestampMilliseconds}ms : ${voltValue}v"
    }

}
